apps/antalmanac/src/components/RightPane/SectionTable/EnrollmentHistoryPopup.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState, useEffect, useMemo } from 'react';
+import { LineChart, Line, CartesianGrid, ResponsiveContainer, XAxis, YAxis, Tooltip, Legend } from 'recharts';
+import { Box, Link, Typography, Skeleton, useMediaQuery } from '@mui/material';
+import { MOBILE_BREAKPOINT } from '../../../globals';
+import { DepartmentEnrollmentHistory, EnrollmentHistory } from '$lib/enrollmentHistory';
+import { isDarkMode } from '$lib/helpers';
+
+export interface EnrollmentHistoryPopupProps {
+    department: string;
+    courseNumber: string;
+}
+
+const EnrollmentHistoryPopup = ({ department, courseNumber }: EnrollmentHistoryPopupProps) => {
+    const [loading, setLoading] = useState(true);
+    const [enrollmentHistory, setEnrollmentHistory] = useState<EnrollmentHistory>();
+    const isMobileScreen = useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT})`);
+
+    const deptEnrollmentHistory = useMemo(() => new DepartmentEnrollmentHistory(department), [department]);
```

Comment:
I think this should be ok, since instances of this class don't really manage their own state and just use a shared static cache. 

From my experience, it's generally paramount to use `useRef` with any sort of external class in React, since `useMemo` can't preserve the class in some instances (like when the component is unrendered, but it still exists?? or something stupid like that in React land ™️ )

One possible usage of classes could be like

```ts
class MyClass {}

function MyComponent() {
  const [open, setOpen] = useState(false);
  const myClass = useRef(new MyClass());

  // e.g. update the class instance as needed
  useEffect(() => {
    myClass.updateState(open)
  }, [open])
}
```
where the class is only created once, and it just updates itself whenever it's necessary based on the component's state. This preserves the external class's state in a way closest to vanilla JS.

TLDR: it's fine, but just be careful using classes in React because it's a garbage technology.